api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -1,173 +1,60 @@
-  --
-  -- Mappings from existing logging systems and formats (or source format for short) must define how severity (or log level) of that particular format corresponds to SeverityNumber
-  -- of this data model based on the meaning given for each range in the above table. [More Information](https://opentelemetry.io/docs/specs/otel/logs/data-model/#mapping-of-severitynumber)
-  --
-  -- [These short names](https://opentelemetry.io/docs/specs/otel/logs/data-model/#displaying-severity) can be used to represent SeverityNumber in the UI
-  --
-  -- In the contexts where severity participates in less-than / greater-than comparisons SeverityNumber field should be used.
-  -- SeverityNumber can be compared to another SeverityNumber or to numbers in the 1..24 range (or to the corresponding short names).
-  , body :: body
-  -- ^ A value containing the body of the log record. Can be for example a human-readable string message (including multi-line) describing the event in a free form or it can be a
-  -- structured data composed of arrays and maps of other values. Body MUST support any type to preserve the semantics of structured logs emitted by the applications.
-  -- Can vary for each occurrence of the event coming from the same source. This field is optional.
-  --
-  -- Type any (functions that use Log should have a Typeclass constraint of (ToValue body) => ...)
-  --    Value of type any can be one of the following:
-  --    - A scalar value: number, string or boolean,
-  --    - A byte array,
-  --    - An array (a list) of any values,
-  --    - A map<string, any>.
-  , resource :: Maybe MaterializedResources
-  -- ^ Describes the source of the log, aka resource. Multiple occurrences of events coming from the same event source can happen across time and they all have the same value of Resource.
-  -- Can contain for example information about the application that emits the record or about the infrastructure where the application runs. Data formats that represent this data model
-  -- may be designed in a manner that allows the Resource field to be recorded only once per batch of log records that come from the same source. SHOULD follow OpenTelemetry semantic conventions for Resources.
-  -- This field is optional.
-  , attributes :: LogAttributes
-  -- ^ Additional information about the specific event occurrence. Unlike the Resource field, which is fixed for a particular source, Attributes can vary for each occurrence of the event coming from the same source.
-  -- Can contain information about the request context (other than Trace Context Fields). The log attribute model MUST support any type, a superset of standard Attribute, to preserve the semantics of structured attributes
-  -- emitted by the applications. This field is optional.
-  }
-  deriving stock (Functor)
+getCurrentTimestamp :: (MonadIO m) => m Timestamp
+getCurrentTimestamp = liftIO $ coerce @(IO TimeSpec) @(IO Timestamp) $ getTime Realtime
```

Comment:
This is used in `emitLogRecord` but not exported by the module. It is a duplicate of `getTimestamp` in `Opentelemetry.Trace.Core` but that cannot be imported as that would create a cyclic module dependency. This is also used in a different way so the documentation is not the same.